### PR TITLE
Added support for MLAS 2.0

### PR DESCRIPTION
**Pull request by Palakila**
Incorporate support for MSAL 2 from sample code in AzureDevOps.InteractiveLogin

### CODE REVIEW

File: src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs [hunk 1]
```diff
@@ -1,39 +1,62 @@
...
 
-            // MSA client ID if you don't have an application ID regsitered with Azure
-            string aadClientId = "872cd9fa-d31f-45e0-9eab-6e460a02d1f1";
+            IEnumerable<IAccount> accounts = await application.GetAccountsAsync();
+            try
+            {
+                result = await application.AcquireTokenSilent(scopes, accounts.FirstOrDefault()).ExecuteAsync();
+
+                // Create authorization header of the form "Bearer {AccessToken}"
+            }
```
Comment (chinarosesz) [inline src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs:50]: I think this comment can be removed.

File: src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs [hunk 2]
```diff
@@ -1,39 +1,62 @@
...
-            string tenant = "microsoft.com";
-
-            string aadAuthority = $"https://login.windows.net/{tenant}";
+            // Initialize the MSAL library by building a public client application
+            application = PublicClientApplicationBuilder.Create(clientId).WithAuthority(authority).WithDefaultRedirectUri().Build();
+            TokenCacheHelper.EnableSerialization(application.UserTokenCache);
 
-            // Fixed static resource Guid for Azure Devops
-            string aadResource = "499b84ac-1321-427f-aa17-267ca6975798";
+            Microsoft.Identity.Client.AuthenticationResult result;
```
Comment (chinarosesz) [inline src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs:42]: >Microsoft.Identity.Client. [](http://example.com/codeflow?start=12&length=26)

Move to using?

File: src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs (hunk 1 above)
Comment (Palakila) [reply src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs:48]: > I think this comment can be removed.

Done. This was left over from code pasted.

File: src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs (hunk 2 above)
Comment (Palakila) [reply src/AzureDevOps.DataIngestor.Sdk/Clients/VssClientHelper.cs]: Removed fully qualified type name  - I had that while I was incorporating the code and had two namespaces with same class name - which is now gone. Tried to use 'using' but doesn't seem like AuthorizationResult is IDisposable thus not required.

**Review by chinarosesz — APPROVED** [review]
:shipit:

**Comment by Palakila** [comment]
Done.

**Pull request merged**